src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -97,12 +97,16 @@ StorageRabbitMQ::StorageRabbitMQ(
                     getContext()->getConfigRef().getString("rabbitmq.username"),
                     getContext()->getConfigRef().getString("rabbitmq.password")))
         , vhost(getContext()->getConfigRef().getString("rabbitmq.vhost", rabbitmq_settings->rabbitmq_vhost.value))
+        , connection_string(rabbitmq_settings->rabbitmq_address)
+        , secure(rabbitmq_settings->rabbitmq_secure.value)
         , semaphore(0, num_consumers)
         , unique_strbase(getRandomName())
         , queue_size(std::max(QUEUE_SIZE, static_cast<uint32_t>(getMaxBlockSize())))
         , milliseconds_to_wait(RESCHEDULE_MS)
 {
     event_handler = std::make_shared<RabbitMQHandler>(loop.getLoop(), log);
+    if (secure)
+        SSL_library_init();
```

Comment:
Why is this function used rather than `OPENSSL_init_ssl()`?

Reply:
Does it suffice because it just works?
Also in https://github.com/CopernicaMarketingSoftware/AMQP-CPP in section `secure connections` it is written
```code
// init the SSL library (this works for openssl 1.1, for openssl 1.0 use SSL_library_init())
OPENSSL_init_ssl(0, NULL);
```